docs/src/main/asciidoc/hibernate-search-guide.adoc

```diff
@@ -0,0 +1,627 @@
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.hibernate.search.elasticsearch.model.Author;
+import org.acme.hibernate.search.elasticsearch.model.Book;
+
+@Path("/library")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class LibraryResource {
+
+    @Inject
+    EntityManager em;
+
+    @PUT
+    @Path("book")
+    @Transactional
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public void addBook(@FormParam("title") String title, @FormParam("authorId") Long authorId) {
+        Author author = Author.findById(authorId);
+        if (author == null) {
+            return;
+        }
+
+        Book book = new Book();
+        book.title = title;
+        book.author = author;
+        book.persist();
+    }
+
+    @DELETE
+    @Path("book/{id}")
+    @Transactional
+    public void deleteBook(@PathParam("id") Long id) {
+        Book book = Book.findById(id);
+        if (book != null) {
+            book.author.books.remove(book);
+            book.delete();
+        }
+    }
+
+    @PUT
+    @Path("author")
+    @Transactional
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public void addAuthor(@FormParam("firstName") String firstName, @FormParam("lastName") String lastName) {
+        Author author = new Author();
+        author.firstName = firstName;
+        author.lastName = lastName;
+        author.persist();
+    }
+
+    @POST
+    @Path("author/{id}")
+    @Transactional
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public void updateAuthor(@PathParam("id") Long id, @FormParam("firstName") String firstName, @FormParam("lastName") String lastName) {
+        Author author = Author.findById(id);
+        if (author == null) {
+            return;
+        }
+        author.firstName = firstName;
+        author.lastName = lastName;
+        author.persist();
+    }
+
+    @DELETE
+    @Path("author/{id}")
+    @Transactional
+    public void deleteAuthor(@PathParam("id") Long id) {
+        Author author = Author.findById(id);
+        if (author != null) {
+            author.delete();
+        }
+    }
+}
+----
+
+Nothing really interesting here: it is just basic Hibernate ORM with Panache operations in a standard JAX-RS service.
+
+In fact, the interesting part is that we will need to add very few elements to make our full text search application working.
+
+== Using Hibernate Search annotations
+
+Let's go back to our entities.
+
+Enabling full text search capabilities for them is as simple as adding some additional annotations.
+
+Let's edit the `Author` entity again to include this content:
```

Comment:
It's a deliberate approach.

I want us to be done with the standard stuff. And then, do a second step with all the Hibernate Search-specific stuff. Otherwise, there's too much noise.

There's no point in starting the application as you won't see anything working anyway.